app/src/main/java/com/github/se/cyrcle/model/parking/ParkingViewModel.kt

```diff
@@ -289,6 +302,64 @@ class ParkingViewModel(
       incrementRadius()
     }
   }
+
+  /**
+   * Adds a report for the currently selected parking and updates the repository.
+   *
+   * This function first verifies that a parking is selected. If no parking is selected, it logs an
+   * error and returns. It then attempts to add the report to the parking repository. Upon
+   * successful addition, the report is evaluated against severity and threshold limits to determine
+   * if a `ReportedObject` should be created and added to the reported objects repository.
+   *
+   * Updates the selected parking's report count and severity metrics and ensures these changes are
+   * reflected in the repository.
+   *
+   * @param report The report to be added, which includes details such as the reason and user ID.
+   * @param user The user submitting the report, required for identifying the reporter.
+   */
+  fun addReport(report: ParkingReport, user: User) {
+    val selectedParking = _selectedParking.value
+    if (selectedParking == null) {
+      Log.e("ParkingViewModel", "No parking selected")
+      return
+    }
+
+    parkingRepository.addReport(
+        report,
+        onSuccess = {
+          // Check thresholds for reporting to `ReportedObjectRepository`
+          // i.e if it's max severity add it to the maximum severity count and check if it doesn't
+          // exceed the maximum severity threshold. Otherwise, check if it doesn't exceed
+          // the standard threshold
+          if ((report.reason.severity == MAX_SEVERITY &&
+              selectedParking.nbMaxSeverityReports >= NB_REPORTS_MAXSEVERITY_THRESH) ||
+              (selectedParking.nbReports >= NB_REPORTS_THRESH)) {
+            reportedObjectRepository.addReportedObject(
+                ReportedObject(
+                    objectUID = selectedParking.uid,
+                    reportUID = report.uid,
+                    reason = ReportReason.Parking(report.reason),
+                    userUID = user.public.userId,
+                    objectType = ReportedObjectType.PARKING),
+                onSuccess = {},
+                onFailure = {})
+          }
+
+          // Update the local reports and parking metrics
+          _selectedParkingReports.update { currentReports ->
+            currentReports?.plus(it) ?: listOf(it)
+          }
+          if (report.reason.severity == MAX_SEVERITY) {
+            selectedParking.nbMaxSeverityReports += 1
+          }
+          selectedParking.nbReports += 1
```

Comment:
is it normal that you add +1 to both nbMaxSeverityReports and nbReports when the report reason is MAX_SEVERITY ?

Reply:
They're separate counters: one checks TOTAL report count, the other how many MAX SEVERITY reports were added. So it makes sense that both should be incremented when we add a maximum severity review

Reply:
Imagine that someone reports a parking for a level 3 issue, and it raises both  selectedParking.nbMaxSeverityReports and selectedParking.nbReports to their respective maximum threshold, which one triggers first ?

Reply:
You make a solid point. However, as of this PR and Sprint, the way "maximum severity" is handled is that less reviews of that severity are required to pass the object into ReportedObjects, so it doesn't make a difference

Reply:
Yes i was just projecting (too much) forward in time, i was imagining this app with a lot of users, and admins checking the reports manually from most severe to least severe. And the issue would be if a severity 3 report would be reported as a severity 1 or 2.